src/main/scala/chisel3/aop/AspectLibrary.scala

```diff
@@ -0,0 +1,47 @@
+// See LICENSE for license details.
+
+package chisel3.aop
+
+import chisel3.RawModule
+import firrtl.options.{OptionsException, RegisteredLibrary, ShellOption}
+
+/** Enables adding aspects to a design from the commandline, e.g.
+  *  sbt> runMain chisel3.stage.ChiselMain --module <module> --with-aspect <aspect>
+  */
+case class AspectLibrary() extends RegisteredLibrary  {
```

Comment:
Any reason why this is a `case class` as opposed to a straight `class`?

Reply:
No, but I don't think it should be extended. Perhaps make it `final` ?

Reply:
```suggestion
final class AspectLibrary() extends RegisteredLibrary  {
```